data-prepper-plugins/key-value-processor/src/test/java/com/amazon/dataprepper/plugins/processor/keyvalue/KeyValueProcessorTests.java

```diff
@@ -97,22 +100,97 @@ void testSingleRegexKvDelimiterKvToObjectKeyValueProcessor() {
 
         final Record<Event> record = getMessage("key1:++:value1&key2:+:value2");
         final List<Record<Event>> editedRecords = (List<Record<Event>>)keyValueProcessor.doExecute(Collections.singletonList(record));
-        LinkedHashMap parsed_message = getLinkedHashMap(editedRecords);
+        final LinkedHashMap parsed_message = getLinkedHashMap(editedRecords);
 
         assertThat(parsed_message.size(), equalTo(2));
         assertThatKeyEquals(parsed_message, "key1", "value1");
         assertThatKeyEquals(parsed_message, "key2", "value2");
     }
 
+    @Test
+    void testBadKeyValueDelimiterRegexKeyValueProcessor() {
+        when(mockConfig.getKeyValueDelimiterRegex()).thenReturn("[");
+        try {
+             keyValueProcessor = new KeyValueProcessor(pluginMetrics, mockConfig);
+             assertThat("Exception should have been thrown.", keyValueProcessor, is(not(anything())));
+        }
+        catch (PatternSyntaxException e) {
+            assertThat(e.getMessage().startsWith("key_value_delimiter_regex"), is(true));
+        }
```

Comment:
What does this error look like? Is it informative / actionable to the user of Data Prepper?

Reply:
The error looks like `"field_delimiter_regex is not a valid regex string"`